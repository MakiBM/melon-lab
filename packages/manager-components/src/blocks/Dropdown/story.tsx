import { storiesOf } from '@storybook/react';
import React from 'react';
import Dropdown from './index';

const options = [{ value: 'one', name: 'One' }, { value: 'two', name: 'Two' }];

storiesOf('Blocks|Dropdown', module)
  .add('Default', () => {
    return <Dropdown options={options} />;
  })
  .add('With label', () => {
    return <Dropdown label="Exchange" options={options} />;
  });
