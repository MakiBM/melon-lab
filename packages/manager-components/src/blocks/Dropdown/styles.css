.dropdown {
  display: flex;
  justify-content: space-between;
  align-items: center;
}

.dropdown__label {
  color: var(--color-grey);
}

.dropdown__wrapper {
  position: relative;
}

.dropdown__wrapper::before {
  content: '';
  position: absolute;
  right: 4px;
  top: 8px;
  width: 0;
  height: 0;
  border-left: 6px solid transparent;
  border-right: 6px solid transparent;
  border-top: 6px solid var(--color-primary-darkest);
}

.dropdown__select {
  appearance: none;
  border: none;
  width: 100%;
  background: transparent;
  font-size: var(--font-size-m);
  border-radius: 0;
  margin-right: var(--space-l);
}

.dropdown__select::-ms-expand {
  display: none;
}

.dropdown__select:focus::-ms-value {
  background-color: transparent;
}
