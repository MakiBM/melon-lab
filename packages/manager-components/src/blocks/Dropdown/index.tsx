import React, { ChangeEventHandler, StatelessComponent } from 'react';

import styles from './styles.css';

export interface DropdownProps {
  options: Array<{
    value: string;
    name: string;
  }>;
  label: string;
  value?: string;
  name: string;
  onChange(value, event);
}

const renderOption = (option, index) => {
  return (
    <option key={index} value={option.value}>
      {option.name}
    </option>
  );
};

const Dropdown: StatelessComponent<DropdownProps> = ({
  options,
  onChange,
  label,
  value,
  name,
}) => {
  const ops =
    options && options.map((item, index) => renderOption(item, index));

  const handleChange = e => {
    onChange({ ...e.target.value }, e);
  };

  return (
    <div className="dropdown">
      <style jsx>{styles}</style>
      {label && <span className="dropdown__label">{label}</span>}
      <div className="dropdown__wrapper">
        <select
          name={name}
          className="dropdown__select"
          onChange={handleChange}
          value={value}
        >
          {ops}
        </select>
      </div>
    </div>
  );
};

export default Dropdown;
