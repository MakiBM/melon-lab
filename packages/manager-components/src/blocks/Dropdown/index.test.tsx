import React from 'react';
import Dropdown from './index';

describe('Dropdown', () => {
  const options = [
    { value: 'one', name: 'One' },
    { value: 'two', name: 'Two' },
  ];
  const defaultElement = <Dropdown options={options} selectedItem={'one'} />;

  it('should render correctly', () => {
    const wrapper = shallow(defaultElement);
    expect(wrapper).toMatchSnapshot();
  });

  it('should be selectable by class "dropdown"', () => {
    expect(shallow(defaultElement).is('.dropdown')).toBe(true);
  });

  it('should mount in a full DOM', () => {
    expect(mount(defaultElement).find('.dropdown').length).toBe(1);
  });

  it('should render to static HTML', () => {
    const getFirstOptionText = render(defaultElement)
      .find('option')
      .first()
      .text();

    expect(getFirstOptionText).toEqual('One');
  });

  it('should pass a selected value to the onChange handler', () => {
    const value = options[0];
    const onChange = jest.fn();
    const wrapper = shallow(
      <Dropdown onChange={onChange} options={options} />,
    );

    expect(wrapper).toMatchSnapshot();
    wrapper.find('select').simulate('change', options[0]);
    expect(onChange).toBeCalledWith(value);
  });
});
