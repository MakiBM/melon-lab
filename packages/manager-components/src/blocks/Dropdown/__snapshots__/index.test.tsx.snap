// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown should pass a selected value to the onChange handler 1`] = `
<div
  className="jsx-__scopedHash dropdown"
>
  <JSXStyle
    css="__scoped"
    styleId="__scopedHash"
  />
  <div
    className="jsx-__scopedHash dropdown__wrapper"
  >
    <select
      className="jsx-__scopedHash dropdown__select"
      onChange={[MockFunction]}
    >
      <option
        key="0"
        value="one"
      >
        One
      </option>
      <option
        key="1"
        value="two"
      >
        Two
      </option>
    </select>
  </div>
</div>
`;

exports[`Dropdown should render correctly 1`] = `
<div
  className="jsx-__scopedHash dropdown"
>
  <JSXStyle
    css="__scoped"
    styleId="__scopedHash"
  />
  <div
    className="jsx-__scopedHash dropdown__wrapper"
  >
    <select
      className="jsx-__scopedHash dropdown__select"
    >
      <option
        key="0"
        value="one"
      >
        One
      </option>
      <option
        key="1"
        value="two"
      >
        Two
      </option>
    </select>
  </div>
</div>
`;
